### ISSUE TITLE
undefined method `say_status'

### ISSUE THREAD

**Issue by trouba** [follmann/middleman-favicon-maker#38]
Hello, I try to use favicon maker with Middleman 4 and Rails 2.3.1. 
When I launch the build, it generates the first favicon in the right directory. After that, I've got the following error:

**/Users/Nicolas/.rvm/gems/ruby-2.3.1/gems/middleman-favicon-maker-3.7/lib/middleman-favicon-maker/extension.rb:34:in block (2 levels) in after_build': undefined method `say_status' for #Middleman::Builder:0x007fae65272340 (NoMethodError)**

Here is my config.rb configuration: 

```

activate :favicon_maker do |f|
    f.template_dir  = "source/"
    f.output_dir  = "source/images/favicon"
    f.icons = {
      "_favicon.png" =>   [
        { icon: "apple-touch-icon-180x180-precomposed.png" },             # Same as apple-touch-icon-57x57.png, for iPhone 6 Plus with @3× display
        { icon: "apple-touch-icon-152x152-precomposed.png" },             # Same as apple-touch-icon-57x57.png, for retina iPad with iOS7.
        { icon: "apple-touch-icon-144x144-precomposed.png" },             # Same as apple-touch-icon-57x57.png, for retina iPad with iOS6 or prior.
        { icon: "apple-touch-icon-120x120-precomposed.png" },             # Same as apple-touch-icon-57x57.png, for retina iPhone with iOS7.
        { icon: "apple-touch-icon-114x114-precomposed.png" },             # Same as apple-touch-icon-57x57.png, for retina iPhone with iOS6 or prior.
        { icon: "apple-touch-icon-76x76-precomposed.png" },               # Same as apple-touch-icon-57x57.png, for non-retina iPad with iOS7.
        { icon: "apple-touch-icon-72x72-precomposed.png" },               # Same as apple-touch-icon-57x57.png, for non-retina iPad with iOS6 or prior.
        { icon: "apple-touch-icon-60x60-precomposed.png" },               # Same as apple-touch-icon-57x57.png, for non-retina iPhone with iOS7.
        { icon: "apple-touch-icon-57x57-precomposed.png" },               # iPhone and iPad users can turn web pages into icons on their home screen. Such link appears as a regular iOS native application. When this happens, the device looks for a specific picture. The 57x57 resolution is convenient for non-retina iPhone with iOS6 or prior. Learn more in Apple docs.
        { icon: "apple-touch-icon-precomposed.png", size: "57x57" },      # Same as apple-touch-icon.png, expect that is already have rounded corners (but neither drop shadow nor gloss effect).
        { icon: "apple-touch-icon.png", size: "57x57" },                  # Same as apple-touch-icon-57x57.png, for "default" requests, as some devices may look for this specific file. This picture may save some 404 errors in your HTTP logs. See Apple docs
        { icon: "favicon-196x196.png" },                                  # For Android Chrome M31+.
        { icon: "favicon-160x160.png" },                                  # For Opera Speed Dial (up to Opera 12; this icon is deprecated starting from Opera 15), although the optimal icon is not square but rather 256x160. If Opera is a major platform for you, you should create this icon yourself.
        { icon: "favicon-96x96.png" },                                    # For Google TV.
        { icon: "favicon-32x32.png" },                                    # For Safari on Mac OS.
        { icon: "favicon-16x16.png" },                                    # The classic favicon, displayed in the tabs.
        { icon: "favicon.png", size: "16x16" },                           # The classic favicon, displayed in the tabs.
        { icon: "favicon.ico", size: "64x64,32x32,24x24,16x16" },         # Used by IE, and also by some other browsers if we are not careful.
        { icon: "mstile-70x70.png", size: "70x70" },                      # For Windows 8 / IE11.
        { icon: "mstile-144x144.png", size: "144x144" },
        { icon: "mstile-150x150.png", size: "150x150" },
        { icon: "mstile-310x310.png", size: "310x310" },
        { icon: "mstile-310x150.png", size: "310x150" }
      ]
    }
  end

```

And the favicon generation doesn't work for me without  f.template_dir and  f.output_dir .

**Comment by iwarner** [comment]
Your Output_Dir should be pointed to the Build folder

i.e.  f.output_dir  = "build/images/favicon"

**Comment by trouba** [comment]
@iwarner I changed 
`f.output_dir  = "source/images/favicon"`
to 
`f.output_dir  = "build/images/favicon"`
but still had the same issue (generating only the first favicon and raising an "undefined 'say_status' error". 

I commented the 'say_status' calling in extension.rb 
```
each_icon do |filepath, template_filepath|
   builder.say_status :create, filepath
end
```
and it generates all icons well (but remove the feedback on 'build').

With more research I've found that say_status has changed ( [https://github.com/middleman/middleman/issues/1492](url) ) and it seems extensions author should use builder.trigger instead:

```
each_icon do |filepath, template_filepath|
  builder.trigger :create, filepath
end
```

I've tested it and it works fine, both for generating icons and giving feedback on build.
